### ISSUE TITLE
Not compatible with Webpack 4

### ISSUE THREAD

**Issue by sandersky** [cascornelissen/svg-spritemap-webpack-plugin#22]
The following error occurs when trying to use this plugin with Webpack 4.

```
/my/project/path/node_modules/svg-spritemap-webpack-plugin/svg-spritemap.js:62
            sourceChunk.files.push(filename);
                        ^

TypeError: Cannot read property 'files' of undefined
    at additionalChunkAssets (/my/project/path/node_modules/svg-spritemap-webpack-plugin/svg-spritemap.js:62:25)
    at SyncHook.eval [as call] (eval at create (/my/project/path/node_modules/tapable/lib/HookCodeFactory.js:17:12), <anonymous>:7:1)
    at SyncHook.lazyCompileHook [as _call] (/my/project/path/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeTree.callAsync.err (/my/project/path/node_modules/webpack/lib/Compilation.js:935:37)
    at _err0 (eval at create (/my/project/path/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:11:1)
    at _async2.default.forEach.err (/my/project/path/node_modules/extract-text-webpack-plugin/dist/index.js:281:11)
    at /my/project/path/node_modules/async/dist/async.js:473:16
    at iteratorCallback (/my/project/path/node_modules/async/dist/async.js:1050:13)
    at /my/project/path/node_modules/async/dist/async.js:958:16
    at _async2.default.forEach.err (/my/project/path/node_modules/extract-text-webpack-plugin/dist/index.js:259:13)
    at /my/project/path/node_modules/async/dist/async.js:473:16
    at iteratorCallback (/my/project/path/node_modules/async/dist/async.js:1050:13)
    at /my/project/path/node_modules/async/dist/async.js:958:16
    at compilation.rebuildModule.err (/my/project/path/node_modules/extract-text-webpack-plugin/dist/index.js:246:26)
    at callback (/my/project/path/node_modules/webpack/lib/Compilation.js:775:35)
    at processModuleDependencies.err (/my/project/path/node_modules/webpack/lib/Compilation.js:797:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by cascornelissen** [comment]
Yeah, I'm aware of this and currently working on `2.0.0` of this plugin which should be compatible with Webpack `4.0.0` and higher. I'm hoping to release it this week, let's keep the issue open to prevent others from reporting the same.

**Comment by cascornelissen** [comment]
People can follow along the [`webpack4`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/tree/webpack4) branch if they want to for now. I'm hoping to release it under `svg-spritemap-webpack-plugin@next` this friday.

**Comment by cascornelissen** [comment]
@sandersky, I've just released `2.0.0-rc.2` (and `2.0.0-rc.1` as well, obviously). Could you report back if this version works correctly for you?

Installing the release candidate can be done by using the following command:

```
npm install extract-text-webpack-plugin@next --save-dev
```

**Comment by sandersky** [comment]
@cascornelissen I assume you meant `npm install svg-spritemap-webpack-plugin@next --save-dev` 😉

I've confirmed this RC works for me though, thanks for getting it out within a week of the Webpack 4 release. 🚀

**Comment by cascornelissen** [comment]
Haha, whoops. Thanks for noticing. Great to hear it works on your end. I'll test it some more during next week and will probably release it as stable before the end of next week. 👍

**Comment by cascornelissen** [comment]
@sandersky, I've released some more fixes into two new RCs. Could you maybe test `2.0.0-rc.4` by running the `npm install svg-spritemap-webpack-plugin@next --save-dev` and building your application? Would love to hear if it's working correctly for you as well.

`2.0.0` stable release is planned for this sunday.

**Comment by cascornelissen** [comment]
The stable release will be postponed until I've solved https://github.com/cascornelissen/svg-spritemap-webpack-plugin/issues/23 in case anyone was wondering ❤️

**Comment by cascornelissen** [comment]
I've just released [`v2.0.0`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v2.0.0) which includes this fix 🚀 

```shell
npm install svg-spritemap-webpack-plugin --save-dev
```
